src/circuit/lookup.rs

```diff
@@ -20,13 +31,12 @@ fn synth<'a, E: Engine, I>(
     assert_eq!(assignment.len(), 1 << window_size);
 
     for (i, constant) in constants.into_iter().enumerate() {
-        let mut cur = assignment[i];
-        cur.negate();
-        cur.add_assign(constant);
-        assignment[i] = cur;
+        assignment[i].add_assign(constant);
+        let cur = assignment[i];
+        // subtract a_i from a_j when the set bits of j contain those of i
         for (j, eval) in assignment.iter_mut().enumerate().skip(i + 1) {
             if j & i == i {
-                eval.add_assign(&cur);
+                eval.sub_assign(&cur); 
```

Comment:
Is this meant to produce the same results as before? Doing a couple examples by hand it is not clear why this should be the case. Are there tests for this?

Reply:
I don't know whether there are specific tests of this method. However, the change is ensured to be a no-op at the constraint level, because @ebfull has tests that check the "hash" of the entire circuit. Any change that altered a constraint would change that hash.

Reply:
Reason it's the same is there's a negate that's deleted above.  There are specific tests for this `synth` function if you scroll down the file.